tabs/pid_tuning.js

```diff
@@ -558,27 +562,18 @@ TABS.pid_tuning.initialize = function (callback) {
 
                 curveContext.lineWidth = 4;
 
-                curveContext.save();
-                curveContext.strokeStyle = '#ff0000';
-                self.rateCurve.draw(rateRoll, rcRate, rcExpo, useSuperExpo, maxAngularVel, curveContext);
-                curveContext.restore();
-
-                curveContext.save();
-                curveContext.translate(0, -4);
-                curveContext.strokeStyle = '#00ff00';
-                self.rateCurve.draw(ratePitch, rcRate, rcExpo, useSuperExpo, maxAngularVel, curveContext);
-                curveContext.restore();
-
-                curveContext.save();
-                curveContext.strokeStyle = '#0000ff';
-                curveContext.translate(0, 4);
-                self.rateCurve.draw(rateYaw, rcRateYaw, rcExpoYaw, useSuperExpo, maxAngularVel, curveContext);
-                curveContext.restore();
+		drawCurve(self.currentRates.roll_rate, self.currentRates.rc_rate, self.currentRates.rc_expo, self.currentRates.superexpo, maxAngularVel, '#ff0000', 0, curveContext);
+
+		drawCurve(self.currentRates.pitch_rate, self.currentRates.rc_rate, self.currentRates.rc_expo, self.currentRates.superexpo, maxAngularVel, '#00ff00', -4, curveContext);
+
+		drawCurve(self.currentRates.yaw_rate, self.currentRates.rc_rate_yaw, self.currentRates.rc_yaw_expo, self.currentRates.superexpo, maxAngularVel, '#0000ff', 4, curveContext);
             }, 0);
         };
 
-        $('.pid_tuning').on('input change', redrawRateCurves).trigger('input');
-        $('.super_expo_checkbox').on('input change', redrawRateCurves).trigger('input');
+        // UI Hooks
+        // curves
+        $('.pid_tuning').on('input change', updateRates);
+        $('.super_expo_checkbox').on('input change', updateRates).trigger('input');
```

Comment:
I don't think there's a need for this listener, since `$('.pid_tuning').on('input change'` should already cover that and it's just calling the same thing twice.